mozci/push.py

```diff
@@ -78,6 +78,24 @@ def backedout(self):
         """
         return bool(self.backedoutby)
 
+    @memoized_property
+    def bugs(self):
+        """The bugs associated with the commits of this push.
+
+        Returns:
+            list: A list of bug IDs.
+        """
+        return set(
+            sum(
+                (
+                    bug["no"]
+                    for changeset in self._hgmo.changesets
+                    for bug in changeset["bugs"]
+                ),
+                [],
+            )
```

Comment:
This looks really confusing to me. Isn't `bug["no"]` already a string? How are we adding that to a list? Looks like the call to `sum` should be removed.

Reply:
This is accumulating the bugs from all the changesets in the push in a single set.
It would be equivalent to:
```Python
bugs = set()
for changeset in self._hgmo.changesets:
    for bug in changeset["bugs"]:  # there should be one per commit, but for some reason "bugs" is a list.
        bugs.add(bug)
```

Reply:
Would you prefer the more explicit version? It's slower, but of course it doesn't matter at all here.

Reply:
I think there's a bug, the call to `sum` fails trying to concatenate a string to a list. Simply removing the `sum` call worked for me. I don't mind the shortened version.

Reply:
To iillustrate, your code is equivalent to:
```python
bugs = []
for changeset in self._hgmo.changesets:
    for bug in changeset["bugs"]:  # there should be one per commit, but for some reason "bugs" is a list.
        bugs.append(bug["no"])

bugs = set(sum(bugs, []))
```

Reply:
You're right, I don't know what I was thinking :P

Reply:
Fixed with bf8d477.